server/events/vcs/github_client.go

```diff
@@ -307,6 +309,40 @@ func (g *GithubClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 	return true, nil
 }
 
+// Check if the Pull Request is locked with :lock emoji
+func (g *GithubClient) PullIsLocked(repo models.Repo, pull models.PullRequest) (bool, error) {
+	statuses, err := g.getRepoStatuses(repo, pull)
+	if err != nil {
+		return false, errors.Wrapf(err, "fetching repo statuses for repo: %s, and pull number: %d", repo.FullName, pull.Num)
+	}
+
+	for _, status := range statuses {
+		if status.GetContext() == SubmitQueueReadinessStatusContext {
+			description := make(map[string]interface{})
+			err := json.Unmarshal([]byte(status.GetDescription()), &description)
+			if err != nil {
+				return false, errors.Wrapf(err, "parsing status description for repo: %s, and pull number: %d", repo.FullName, pull.Num)
+			}
+
+			if waitingList, ok := description["waiting"]; ok {
+				waitingList, ok := waitingList.([]interface{})
+				if !ok {
+					return true, fmt.Errorf("error retrieving waiting status from description: %s for repo: %s, pull number: %d", status.GetDescription(), repo.FullName, pull.Num)
```

Comment:
Can you leave a comment as to why you are returning true here? I'm assuming it's because we want to be conservative in the case we can't parse the schema but then shouldn't we return true on line 324 as well?


Also, we should just standardize on only `errors.Wrapf()` as mentioned here:
https://github.com/lyft/atlantis/blob/release-v0.17.1-lyft.1/CONTRIBUTING.md#errors

Reply:
I think it's best if we return true only if we can successfully parse the SQ status and find the lock key. So, I've changed the code to return true only if we find the lock key. And I have updated the error message as well.